libraries/AR_Motors/AP_MotorsUGV.h

```diff
@@ -221,9 +223,25 @@ class AP_MotorsUGV {
     float   _throttle_factor[AP_MOTORS_NUM_MOTORS_MAX];
     float   _steering_factor[AP_MOTORS_NUM_MOTORS_MAX];
     float   _lateral_factor[AP_MOTORS_NUM_MOTORS_MAX];
-    uint8_t   _motors_num;
+    uint8_t    _motors_num;
+    uint32_t    _last_set_throttle_call;   // used by esc_backward_unlock logic
+    float    _esc_backward_unlock_simulated_signal;  // signal sent to esc to simulate backward unlock
+    bool    _esc_backward_unlock_active;  // esc unlock process in active     
+    UnlockState    _esc_backward_unlock_state;  // state machine status for backward unlock logic.
```

Comment:
you could shove these in a struct.
```suggestion

    // backward unlock logic variables
    struct {
        uint32_t    last_set_throttle_call;
        float    simulated_signal;  // signal sent to esc to simulate backward unlock
        bool   active;  // esc unlock process in active     
        UnlockState    state;  // state machine status
    }  _esc_backwards_unlock;
```